test/cpp/using-alias.cpp

```diff
@@ -0,0 +1,9 @@
+/** Type alias */
```

Comment:
> * Using `autotype` directive. In fact it seems we skipped this one before somehow, but I guess there's no better time than now :)

I had this at first, but using an array as arguments in the yaml file resultet in the text and html tests missing. Any hints to why are appreciated.

> * More complicated templates! I'm thinking multiple parameter templates, but also variadic and nested ones. I'm not sure those are handled correctly with this simple implementation, but haven't tried it so far. You can have a look at the templates tests and `DocCursor._get_template_line`, check if there's any common logic to it (hopefully yes).

Fair, I can add something that is a bit more involved.

> * `using namedstruct = struct named { int foo; };` and `using anonstruct = struct { int foo; };` are also missing. A quick test seems to yield this for the anonymous case:

As commented [here](https://github.com/jnikula/hawkmoth/pull/226#issuecomment-1828569891), I would explicitly leave that out since it does not make sense to me to support it.

Reply:
> > * Using `autotype` directive. In fact it seems we skipped this one before somehow, but I guess there's no better time than now :)
> 
> I had this at first, but using an array as arguments in the yaml file resultet in the text and html tests missing. Any hints to why are appreciated.

Hmm, can't say that I know what you mean. Maybe push the commits with the tests and we can have a look then ;)

> > * `using namedstruct = struct named { int foo; };` and `using anonstruct = struct { int foo; };` are also missing. A quick test seems to yield this for the anonymous case:
> 
> As commented [here](https://github.com/jnikula/hawkmoth/pull/226#issuecomment-1828569891), I would explicitly leave that out since it does not make sense to me to support it.

Jani seems to be ok with ignoring it too, so I guess that's fine.

I wonder if we should put in the tests anyway just so we know we don't suddenly crash if a user decides to feed it such code anyway. Say with a clang update or some other change... Overthinking much perhaps? @jnikula